libbeat/dashboards/modify_json.go

```diff
@@ -46,20 +46,16 @@ func ReplaceIndexInIndexPattern(index string, content common.MapStr) common.MapS
 		return content
 	}
 
-	objects, ok := content["objects"].([]interface{})
+	objects, ok := content["objects"].([]common.MapStr)
```

Comment:
It looks like this `ReplaceIndexInIndexPattern` function is used by a few code paths. There is one case where it is used with the internally generated index pattern (this probably contains `[]common.MapStr`) and another case where it operates on data produced directly from a `json.Unmarshal` (this would have `[]interface`).

So perhaps this needs to handle both cases? Take a look at the callers `ReplaceIndexInIndexPattern`.

Reply:
Ah, you're right. I don't think we usually use the direct file import functionality since https://github.com/elastic/beats/pull/10478, but it's still there. I pushed an update to make both cases work.